daemon/src/daemon.rs

```diff
@@ -41,6 +41,14 @@ pub enum DocMessage {
     RemoveFile {
         file_path: String,
     },
+    CreateFile {
```

Comment:
We could group removal, creation and update to a "sub enum", similar to `FromEditor`, wdyt? Or something more dynamic:
```rust
enum FileOperationKind {
	Creation,
	Removal,
	Update,
}
pub enum DocMessage {
...
	FileOperation {
		kind: FileOperationKind,
		file_path: String,
		content: Option<String>,
	}
...
}
```